rust/src/pgsql/pgsql.rs

```diff
@@ -480,30 +515,36 @@ impl PgsqlState {
                     );
                     start = rem;
                     SCLogDebug!("Response is {:?}", &response);
-                    if let Some(state) = self.response_process_next_state(&response, flow) {
+                    let new_state = self.response_process_next_state(&response, flow);
+                    if let Some(state) = new_state {
                         self.state_progress = state;
-                    };
-                    let tx_completed = self.is_tx_completed();
-                    let curr_state = self.state_progress;
+                    }
                     if let Some(tx) = self.find_or_create_tx() {
-                        if curr_state == PgsqlStateProgress::DataRowReceived {
-                            tx.incr_row_cnt();
-                        } else if curr_state == PgsqlStateProgress::CommandCompletedReceived
-                            && tx.get_row_cnt() > 0
-                        {
-                            // let's summarize the info from the data_rows in one response
-                            let dummy_resp =
-                                PgsqlBEMessage::ConsolidatedDataRow(ConsolidatedDataRowPacket {
-                                    identifier: b'D',
-                                    row_cnt: tx.get_row_cnt(),
-                                    data_size: tx.data_size, // total byte count of all data_row messages combined
-                                });
-                            tx.responses.push(dummy_resp);
-                            tx.responses.push(response);
-                        } else {
-                            tx.responses.push(response);
-                            if tx_completed {
-                                tx.tx_state = PgsqlTransactionState::ResponseDone;
+                        if tx.tx_res_state == PgsqlTxProgress::TxInit {
+                            tx.tx_res_state = PgsqlTxProgress::TxReceived;
+                        }
+                        if let Some(state) = new_state {
+                            if state == PgsqlStateProgress::DataRowReceived {
+                                tx.incr_row_cnt();
+                            } else if state == PgsqlStateProgress::CommandCompletedReceived
+                                && tx.get_row_cnt() > 0
+                            {
+                                // let's summarize the info from the data_rows in one response
+                                let dummy_resp = PgsqlBEMessage::ConsolidatedDataRow(
+                                    ConsolidatedDataRowPacket {
+                                        identifier: b'D',
+                                        row_cnt: tx.get_row_cnt(),
+                                        data_size: tx.data_size, // total byte count of all data_row messages combined
+                                    },
+                                );
+                                tx.responses.push(dummy_resp);
```

Comment:
Can `responses` grow forever ?

Reply:
Based on [PGSQL's limits doc](https://www.postgresql.org/docs/current/limits.html), and the fact that a `SELECT * FROM` could return all rows in a table, not _forever_, but for a while. Should we come up with a configurable limit?

Reply:
I think it is wise to have a limit